### PR TITLE
[SM-193] Access Policy

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add database structure for Access Policies

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by Thomas-Avery — APPROVED** [review]
Great work on this. I left a comment about updating the commented out code with a `TODO` on when to turn on access policy checks, but I'm good either way.

**Review by Thomas-Avery — APPROVED** [review]
LGTM

File: src/Core/Entities/AccessPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        Id = CoreHelpers.GenerateComb();
+    }
+}
+
+public class UserProjectAccessPolicy : BaseAccessPolicy
+{
+    public Guid? OrganizationUserId { get; set; }
+    public Guid? GrantedProjectId { get; set; }
+}
```
Comment (MGibson1) [inline src/Core/Entities/AccessPolicy.cs:52]: This feels like a weird way to do it. I had expected `AccessPolicy` to be an interface that was explicitly implemented in each of the `___AccessPolicy` types to hide irrelevant properties

```suggestion
public interface IAccessPolicy : ITableObject<Guid>
{
    Guid Id { get; set; }

    // Object to grant access from
    Guid? OrganizationUserId { get; set; }
    Guid? GroupId { get; set; }
    Guid? ServiceAccountId { get; set; }

    // Object to grant access to
    Guid? GrantedProjectId { get; set; }
    Guid? GrantedServiceAccountId { get; set; }

    // Access
    bool Read { get; set; }
    bool Write { get; set; }

    DateTime CreationDate { get; set; }
    DateTime RevisionDate { get; set; }

    void SetNewId();
}

public class UserProjectAccessPolicy : IAccessPolicy
{
    public Guid? OrganizationUserId { get; set; }
    public Guid? GrantedProjectId { get; set; }

    public bool Read { get; set; }
    public bool Write { get; set; }

    Guid IAccessPolicy.GroupId { get; set; }
    Guid IAccessPolicy.ServiceAccountId { get; set; }
    Guid IAccessPolicy.GrantedServiceAccountId { get; set; }

    public void SetNewId()
    {
        Id = CoreHelpers.GenerateComb();
    }
}
```

Comment (Hinton) [reply src/Core/Entities/AccessPolicy.cs:52]: I don't think that will play well with EF and I don't particularly feel like digging into potentially finding a solution for it :(.

**Pull request merged**